.github/workflows/codeql.yml

```
@@ -0,0 +1,32 @@
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: typescript
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v3
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)